### ISSUE TITLE
DEP: Finish deprecation in issubdtype

### ISSUE THREAD

**Issue by rfezzani** [numpy/numpy#14619]
I noticed that the `issubdtype` function returns inconsistent output for `float32`, `int32` and `int8`. I did not investigated more to see if this affects other dtypes...

### Reproducing code example:

```python
import numpy as np

np.issubdtype('float32', np.float64)  # False
np.issubdtype('float32', np.double)  # False
np.issubdtype('float32', np.float)  # True
np.issubdtype('float32', 'float64')  # True

np.issubdtype('int32', np.int64)  # False
np.issubdtype('int32', 'int64')  # True

np.issubdtype('int8', np.int64)  # False
np.issubdtype('int8', 'int64')  # True

np.issubdtype('int8', np.int32)  # False
np.issubdtype('int8', 'int32')  # True

np.issubdtype('int8', np.int16)  # False
np.issubdtype('int8', 'int16')  # True

```

### Numpy/Python version information:

`1.16.4 3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0]`

**Comment by seberg** [comment]
You should be seeing a FutureWarning for exactly that reason when you run your code above. That FutureWarning has been there for a few years, so it could probably be switched over (there should be a tracking issue on that htough).
Although, to be honest, I am not quite sure what this function is useful for in either case...

**Comment by rfezzani** [comment]
Hi @seberg, thank you for your quick answer. `issubdtype` is used for example in [`skimage.dtype.convert`](https://github.com/scikit-image/scikit-image/blob/e28ccfbd2abf818c1d972bd6b89c873be2f959b3/skimage/util/dtype.py#L126): I observed this bug while debugging this function :smile:

**Comment by seberg** [comment]
Probably even works there, although I expect this can also be just a `dtype1 == dtyp2` check, since probably in all other cases it ends up being the same. Although that will miss byteswapped dtypes. OTOH, right now it just ignores that it might have to byteswap...

**Comment by seberg** [comment]
Ah sorry, the comment above specifically wants to support `np.floating`, so it actually makes sense here...

**Comment by seberg** [comment]
Sorry for the spam, but your code seems fine in the sense that it calls `np.dtype` on such strings in any case? So that it cannot run into the issue. Although the promise of supporting `np.floating` seems to be broken just a few lines below it, so not sure what the use would be.

**Comment by eric-wieser** [comment]
@seberg, any reason this is marked :"Close?"? 

Milestoning as 1.19, since it's been slightly less than two years since this future warning was introduced in 1.14

**Comment by seberg** [comment]
No, its probably part of Matti's deprecation issue and the original issue was not to "finish" it, but the inconsistency that will be fixed after it is finished.

**Comment by seberg** [comment]
The deprecation was finalized in gh-15773 and thus is included as part of 1.19.
